tests/network/expose.go

```diff
@@ -825,3 +796,23 @@ func generateHelloWorldServer(vmi *v1.VirtualMachineInstance, testPort int, prot
 		&expect.BExp{R: console.RetValue("0")},
 	}, 60)).To(Succeed())
 }
+
+func createAndWaitForJobToSucceed(jobFactory func(host, port string) *batchv1.Job, namespace, ip, port, viaMessage string) error {
+	By(fmt.Sprintf("Starting a job which tries to reach the VMI via the %s", viaMessage))
+	jobInstance := jobFactory(ip, port)
+	jobInstance, err := kubevirt.Client().BatchV1().Jobs(namespace).Create(context.Background(), jobInstance, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	By("Waiting for the job to report a successful connection attempt")
+	return job.WaitForJobToSucceed(jobInstance, time.Duration(120)*time.Second)
+}
+
+func runJobAgainstService(svc *k8sv1.Service, namespace string, jobFactory func(host, port string) *batchv1.Job) {
+	serviceIPs := svc.Spec.ClusterIPs
+	for ipOrderNum, ip := range serviceIPs {
+		servicePort := strconv.Itoa(int(svc.Spec.Ports[0].Port))
+		Expect(createAndWaitForJobToSucceed(jobFactory, namespace, ip, servicePort, fmt.Sprintf("%d ClusterIP", ipOrderNum+1))).To(Succeed())
```

Comment:
@RamLavi here too, the function asserts. do you want to fix it as well, to return error?

Reply:
for sure. DONE